util/coinach.ts

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env node
```

Comment:
I think the shebang here would be updated as well:
```suggestion
#!/usr/bin/env ts-node
```
I think this also invites a conversation if this file should be executable to match the previous Python script usage, or if we should instead drop the shebang at the top of each TypeScript util file (drop their executable nature) and instead expect them all to run from `npm run`/`yarn` commands.

Reply:
I'm not sure I have strong opinions here.  I think mostly `npm run` commands are there to be easier for commonly run commands (e.g. find-translations) or things that are run by build scripts (e.g. the coverage report, just to have a layer of abstraction so that it's easy to know "ok I run X command to do the same thing as the build step").  Other things in util are probably used once a patch by 1-2 people tops, and so it doesn't feel like it needs the same level of accessibility in some ways.

Is there a reason not to have the shebang?

Reply:
> Is there a reason not to have the shebang?

Strictly, I think no. I think the shebang, regardless if the file is committed as an executable or not (I don't think it is here, but I can't firmly tell?), is okay to "just have". I would want to confirm that running this file as an executable (`./coinach.ts`) works as expected; I guess at that point the expectation there would be someone installing TypeScript globally, since I doubt `ts-node` is on your path if it's only in your `node_modules/` directory (something that should be documented accordingly, in an ideal world).

I guess this is more a question of "Should these files have two entrypoints?" which I think the answer is going to always want to be no, so we should firm up our single entrypoint for these, which is diverging from the place where entrypoints live. Sorry, this may be a lot of thinking for something kind of pointless, but I guess this is mainly "Should we close the loop on this pattern and expect people to only run from the commands in `package.json`"?

Reply:
Maiko also made this suggestion on another patch: https://github.com/quisquous/cactbot/pull/2603#issuecomment-802670848 in terms of having a single npm entrypoint for all the scripts (with a chooser), which I think would prevent proliferation of a million gen_x_data entry points in `package.json`

Regardless, I'm fine with removing the shebangs for now in favor of moving towards a more consistent approach in the future.  I don't think I call any scripts directly like that.

Reply:
I think I'm of the same opinion, mostly because I think that an implicit goal of converting things to TypeScript should be that you can do more without having to do external installs. For instance, a pain point around Python is that you have to install it separately from the repository. Ideally, we could limit the dev world down to only needing to install Node and then everything that the repo expects would live inside node_modules and can be called from there. Allowing these files to be called directly would mean that the end user is installing TypeScript globally, with a potentially separate version.

Reply:
Okay, I'll remove the shebang later.